app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
+
+
+def create_person_list(people: list[dict]) -> list:
+    for person in people:
+        _ = Person(name=person.get("name"), age=person.get("age"))
+
+    for person in people:
+        person_name = person.get("name")
+
+        if (wife := person.get("wife")) is not None:
```

Comment:
There is no need for `is not None`, the assignment expression would provide that check